rehttp_server_post17_test.go

```diff
@@ -0,0 +1,136 @@
+//go:build go1.7
+// +build go1.7
+
+package rehttp
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestTransport_RoundTripTimeouts(t *testing.T) {
+	attemptCountMu := sync.Mutex{}
+	attemptCount := 0
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		attemptCountMu.Lock()
+		attemptCount++
+		attemptCountMu.Unlock()
+		time.Sleep(time.Millisecond * 100)
+		w.WriteHeader(http.StatusTooManyRequests)
+	}))
+	defer ts.Close()
+
+	tr := NewTransport(
+		http.DefaultTransport,
+		RetryAll(RetryMaxRetries(3), RetryAny(
+			RetryStatuses(http.StatusTooManyRequests), // retry 429s
+			func(attempt Attempt) bool { // retry context deadline exceeded errors
+				return attempt.Error != nil && attempt.Error == context.DeadlineExceeded // errors.Is requires go 1.13+
+			})),
+		ConstDelay(time.Duration(0)),
+	)
+	tr.PerAttemptTimeout = time.Millisecond * 10 // short timeout
+
+	client := http.Client{
+		Transport: tr,
+	}
+
+	req, err := http.NewRequest(http.MethodGet, ts.URL, nil)
+	if err != nil {
+		t.Errorf("error creating request: %s", err)
+	}
+
+	_, err = client.Do(req)
+	time.Sleep(time.Millisecond * 100) // let the server finish sleeping
```

Comment:
Not a fan of having sleeps in there as this is so often a source of flaky tests, but this might get a bit tricky to do otherwise (e.g. with a channel, waiting for it to be closed here). I'm fine keeping it as-is if the alternative is too complex.